### PR TITLE
preprocess owid csv

### PR DESCRIPTION
**Pull request by hanlu09205**
This only clean the CSV we got from OWID. 

Verified by running the script and check output.

---
This change is [Reviewable](https://reviewable.io/reviews/datacommonsorg/data/362)

### CODE REVIEW

File: scripts/covid_owid_project/preprocess_csv.py [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                            lineterminator='\n')
+    gcontext = ssl.SSLContext()                         
+    with urllib.request.urlopen(
+            'https://raw.githubusercontent.com/owid/covid-19-data/master/public/data/owid-covid-data.csv', context=gcontext) as response:
+        reader = csv.DictReader(io.TextIOWrapper(response))
+
+        writer.writeheader()
+        for row_dict in reader:
+            # Skip invalid country ISO code.
+            if len(row_dict['iso_code']) != 3 :
```
Comment (pradh) [inline scripts/covid_owid_project/preprocess_csv.py:45]: Might be good to restrict it to the ones listed here: https://github.com/datacommonsorg/data/blob/master/util/alpha2_to_dcid.py#L16

Unless you've confirmed there aren't invalid 3-letter country codes?

**Comment by pradh** [comment]
Please consider adding a README, like the other imports...

**Review by hanlu09205 — COMMENTED** [review]
Definitely will do.

> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/data/362#-:-MQt05X54JPqMWdy4lLf:b-cjymrt)* status: 0 of 1 files reviewed, 1 unresolved discussion (waiting on @pradh)

___
*[scripts/covid_owid_project/preprocess_csv.py, line 45 at r1](https://reviewable.io/reviews/datacommonsorg/data/362#-MQt00gYa9qBXzxBaDHp-r1-45:-MQt0F393NNaaswmJ3Ao:b-896fix) ([raw file](https://github.com/datacommonsorg/data/blob/e0bae73f8cc51b7cfb5df9eac538c32e9997cb65/scripts/covid_owid_project/preprocess_csv.py#L45)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

Might be good to restrict it to the ones listed here: https://github.com/datacommonsorg/data/blob/master/util/alpha2_to_dcid.py#L16

Unless you've confirmed there aren't invalid 3-letter country codes?
</blockquote></details>

Done.

**Review by pradh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/data/362)* status: 0 of 1 files reviewed, 4 unresolved discussions (waiting on @hanlu09205 and @pradh)

___
*[scripts/covid_owid_project/preprocess_csv.py, line 49 at r2](https://reviewable.io/reviews/datacommonsorg/data/362#-MQt3glIETqIfUEJdFVM:-MQt3glIETqIfUEJdFVN:bqj3l02) ([raw file](https://github.com/datacommonsorg/data/blob/ee63d8858aa27b1e4401da1f1488457e9eebf17b/scripts/covid_owid_project/preprocess_csv.py#L49)):*
> ```
> 'country/%s' % row_dict['iso_code']
> ```

nit: maybe use a variable place_dcid for this, and use it below too?
___
*[scripts/covid_owid_project/preprocess_csv.py, line 49 at r2](https://reviewable.io/reviews/datacommonsorg/data/362#-MQt3rb2C3xa-3KXHGbY:-MQt3rb2C3xa-3KXHGbZ:bim7o9v) ([raw file](https://github.com/datacommonsorg/data/blob/ee63d8858aa27b1e4401da1f1488457e9eebf17b/scripts/covid_owid_project/preprocess_csv.py#L49)):*
> ```
> in COUNTRY_MAP.values():
> ```

While this is not performance sensitive code, still feels wasteful to do a O(N^2) loop here. How about we use a set?

country_set = set(COUNTRY_MAP.values())
___
*[scripts/covid_owid_project/preprocess_csv.py, line 53 at r2](https://reviewable.io/reviews/datacommonsorg/data/362#-MQt4fHDDedWfeXuw0Hm:-MQt4fHDDedWfeXuw0Hn:b-hragnr) ([raw file](https://github.com/datacommonsorg/data/blob/ee63d8858aa27b1e4401da1f1488457e9eebf17b/scripts/covid_owid_project/preprocess_csv.py#L53)):*
> ```
> '%s' % row_dict['date']
> ```

for my understanding why not just row_dict['date']?

**Review by hanlu09205 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/data/362)* status: 0 of 1 files reviewed, 3 unresolved discussions (waiting on @pradh)

___
*[scripts/covid_owid_project/preprocess_csv.py, line 49 at r2](https://reviewable.io/reviews/datacommonsorg/data/362#-MQt3glIETqIfUEJdFVM:-MQtMqix0oADfQTV7upT:b-896fix) ([raw file](https://github.com/datacommonsorg/data/blob/ee63d8858aa27b1e4401da1f1488457e9eebf17b/scripts/covid_owid_project/preprocess_csv.py#L49)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

> ```
> 'country/%s' % row_dict['iso_code']
> ```

nit: maybe use a variable place_dcid for this, and use it below too?
</blockquote></details>

Done.
___
*[scripts/covid_owid_project/preprocess_csv.py, line 49 at r2](https://reviewable.io/reviews/datacommonsorg/data/362#-MQt3rb2C3xa-3KXHGbY:-MQtMs2-6yYVmW2KIGmH:b-896fix) ([raw file](https://github.com/datacommonsorg/data/blob/ee63d8858aa27b1e4401da1f1488457e9eebf17b/scripts/covid_owid_project/preprocess_csv.py#L49)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

> ```
> in COUNTRY_MAP.values():
> ```

While this is not performance sensitive code, still feels wasteful to do a O(N^2) loop here. How about we use a set?

country_set = set(COUNTRY_MAP.values())
</blockquote></details>

Done.
___
*[scripts/covid_owid_project/preprocess_csv.py, line 53 at r2](https://reviewable.io/reviews/datacommonsorg/data/362#-MQt4fHDDedWfeXuw0Hm:-MQtMzPpBdC6G7VMugKO:bb5vd4o) ([raw file](https://github.com/datacommonsorg/data/blob/ee63d8858aa27b1e4401da1f1488457e9eebf17b/scripts/covid_owid_project/preprocess_csv.py#L53)):*
<details><summary><i>Previously, pradh (Prashanth R) wrote…</i></summary><blockquote>

> ```
> '%s' % row_dict['date']
> ```

for my understanding why not just row_dict['date']?
</blockquote></details>

Good point. 
Because I was copying 
'Date':
                    '%s-%s-%s' % (row_dict['date'][:4], row_dict['date'][4:6],
                                  row_dict['date'][6:]),
Done.

**Review by pradh — APPROVED** [review]
> Reviewed 1 of 1 files at r3.
> *[Reviewable](https://reviewable.io/reviews/datacommonsorg/data/362)* status: all files reviewed, 3 unresolved discussions (waiting on @pradh)

**Pull request merged**